packages/apps/human-app/frontend/src/api/fetcher.ts

```diff
@@ -71,6 +71,46 @@ export function createFetcher(defaultFetcherConfig?: {
   options?: RequestInit | (() => RequestInit);
   baseUrl: FetcherUrl | (() => FetcherUrl);
 }) {
+  let isRefreshing = false;
+  let refreshPromise: Promise<SignInSuccessResponse | undefined> | null = null;
+
+  async function refreshToken(): Promise<
```

Comment:
Do we need to have it inside of `createFetcher` or can move outside?
We don't use `baseUrl` or `options`, `refreshToken` relies on `apiClient.fetcher` to make its request (which looks like circular dependency).

Would be nice to move it outside and to separate file in order to remove circular dependency.

Reply:
Yes, that's actually something I wanted to get rid of using plain fetch, good catch!